### PR TITLE
Rack::Throttle::Burst

### PR DESCRIPTION
**Pull request by emiellohr**
Hi, i've written a method to allow a limited number of requests per sliding window.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,24 +42,29 @@ Examples
...
     use Rack::Throttle::Interval
-    
+
     run lambda { |env| [200, {'Content-Type' => 'text/plain'}, "Hello, world!\n"] }
 
 ### Enforcing a minimum 3-second interval between requests
 
     use Rack::Throttle::Interval, :min => 3.0
 
+### Allowing a maximum of 10 requests per 5 seconds, sliding window and bann for 1 minute
```
Comment (GuyPaddock) [inline README.md]: Ban is misspelled

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-0.4.0
+0.4.4
```
Comment (GuyPaddock) [inline VERSION]: Not sure this PR needs to control the version -- that should probably be handled by the maintainers.

File: lib/rack/throttle/burst.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    # @param  [#call]                  app
+    # @param  [Hash{Symbol => Object}] options
+    # @option options [Integer] :sliding_time_window  (5)
+    # @option options [Integer] :max_per_window       (10)
+    # @option options [Integer] :banned_secs          (60)
+    # @option options [Symbol] :throttle              (:on)
+    def initialize(app, options = {})
+      options = DEFAULTS.merge(options)
+      super
+      logger.debug "[Rack::Throttle::Burst] init_options: #{options.map{|k,v| "#{k}=#{v}"}.join(',')}"
```
Comment (GuyPaddock) [inline lib/rack/throttle/burst.rb]: This should be in a block to reduce overhead when debug logging is not enabled.

**Comment by FreekingDean** [comment]
Hi @emiellohr I know this will be two years old in Feb. (:birthday:) and I know you are not super active in Github these days. I see this PR as 3 kinda separate PR's:
- Burst
- Ban
- NullLogger

I'd like to keep each throttler fairly inline with the others. If we implement "ban" on one we should use it on them all. I am going to keep this open for a few days longer, but if you don't respond (Which I know your not on here anymore so I'm guessing you wont :)) I will close this and open a feature-request issue outlining what you were trying to accomplish here, as well as the code so someone else can hopefully take it over!

**Comment by FreekingDean** [comment]
Going to close this, but someone please re-open if they'd like to take it over :)

**Pull request closed without merging**